File: typescript-client-sdk/src/sdk/models/shared/projectstatusresponsepayload.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { ProjectRunStatusEnum } from "./projectrunstatusenum";
import { Expose, Transform } from "class-transformer";


// ProjectStatusResponsePayload
/** 
 * Response format returned by the getRunStatus endpoint
**/
export class ProjectStatusResponsePayload extends SpeakeasyBase {
  @SpeakeasyMetadata()
  @Expose({ name: "elapsedTime" })
  elapsedTime: number;

  @SpeakeasyMetadata()
  @Expose({ name: "endTime" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  endTime: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "projectId" })
  projectId: string;

  @SpeakeasyMetadata()
  @Expose({ name: "projectVersion" })
  projectVersion: number;

  @SpeakeasyMetadata()
  @Expose({ name: "runId" })
  runId: string;

  @SpeakeasyMetadata()
  @Expose({ name: "runUrl" })
  runUrl: string;

  @SpeakeasyMetadata()
  @Expose({ name: "startTime" })
  @Transform(({ value }) => new Date(value), { toClassOnly: true })
  startTime: Date;

  @SpeakeasyMetadata()
  @Expose({ name: "status" })
  status: ProjectRunStatusEnum;

  @SpeakeasyMetadata()
  @Expose({ name: "traceId" })
  traceId: string;
}

File: go-client-sdk/pkg/models/shared/projectstatusresponsepayload.go
package shared

import (
	"time"
)

// ProjectStatusResponsePayload
// Response format returned by the getRunStatus endpoint
type ProjectStatusResponsePayload struct {
	ElapsedTime    float64              `json:"elapsedTime"`
	EndTime        time.Time            `json:"endTime"`
	ProjectID      string               `json:"projectId"`
	ProjectVersion int                  `json:"projectVersion"`
	RunID          string               `json:"runId"`
	RunURL         string               `json:"runUrl"`
	StartTime      time.Time            `json:"startTime"`
	Status         ProjectRunStatusEnum `json:"status"`
	TraceID        string               `json:"traceId"`
}
